Updated build for Windows including tests

bagge opened the pull request:
I have added a Visual Studio 2013 solution in WinSpecific\vs12 and also integrated the build of the tests into this.
Some minor tweaks of the code in the tests were needed to make it build on windows. These use #ifdef WIN32. Also a minor incompatibility with select() on windows was found when running the tests and a workaround for this has been added in lcm.c.

I have not removed the existing VS2008 files yet as I was unsure if you wanted to do this. I don't know if they are still updated but since the new files are entirely located to WinSpecific\vs12 the old ones can be easily removed when needed.

bagge commented:
Just noticed that the change in run_client_server_test.py would not work with the other clients on linux, so I reverted that part of the change in 968666a.

ashuang commented on test/c/eventlog_test.cpp:
```diff
@@ -6,16 +6,30 @@
 
 TEST(LCM_C, EventLogBasic) {
     // Open and close a log file.
+#ifndef WIN32
```
To minimize the use if #ifdef, suggest adding the following functions (or similar) to test/c/common.c:

    char* make_tmpnam() {
    #ifndef WIN32
      return tmpnam(NULL);
    #else
      return _tempnam(NULL, NULL);
    #endif
    }

    void free_tmpnam(char* fname) {
    #ifdef WIN32
      free(fname);
    #endif
    }

Then modify eventlog_test.cpp to always call make_tmpnam() and free_tmpnam() regardless of platform.

ashuang commented on test/run_client_server_tests.py:
```diff
@@ -22,17 +22,30 @@ def main():
     # Start the test server
-    if not os.path.exists("c/server"):
+    serverName = "c/server"
+    if os.name == "nt":
```
instead of this, let's start using os.path.join:

    serverName = os.path.join("c", "server")

ashuang commented on test/run_client_server_tests.py:
```diff
@@ -22,17 +22,30 @@ def main():
     for name, prog in to_test.items():
-        client_status = subprocess.Popen(prog, shell=True).wait()
-        test_passed[name] = client_status == 0
+        if os.name == "nt":
```
instead of this, suggest using os.path.join() in the definition of to_test

ashuang commented on lcm/lcm.c:
```diff
@@ -233,6 +233,13 @@ lcm_handle_timeout (lcm_t *lcm, int timeout_milis)
   timeout.tv_usec = (timeout_milis % 1000) * 1000;
 
+// Select in Win32 does not return error on negative timeouts.
+#ifdef WIN32
```
this seems like a reasonable thing to do regardless of platform. suggest removing the #ifdef and run the check on all platforms.

ashuang commented:
Looks good, thanks for the pull request. Let me know if you have a chance to address those comments, otherwise we can merge in first and then address them later.

bagge commented:
Thanks for the feedback, I agree with your comments. I'll try to address them soon.

bagge commented:
Ok, I have rebased the pull request and addressed your comments. I have tested the build and the tests on Windows and Ubuntu. Let me know if you have any more comments!

ashuang commented:
Looks great, thanks!
